src/validation.cpp

```diff
@@ -1870,6 +1870,10 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
     txdata.reserve(block.vtx.size()); // Required so that pointers to individual PrecomputedTransactionData don't get invalidated
     for (unsigned int i = 0; i < block.vtx.size(); i++)
     {
+        if (ShutdownRequested()) {
```

Comment:
@TheBlueMatt are you saying this might crash if it aborts during the genesis block (and only the genesis block)? If so, I will add an extra check to exclude this exceptional (and unlikely) situation.